driver/src/contracts/dfusion.rs

```diff
@@ -1,17 +1,30 @@
 #[cfg(test)]
 extern crate mock_it;
 
-use web3::contract::{Contract, Options};
+use web3::contract::Options;
 use web3::futures::Future;
-use web3::types::{Address, BlockId, H256, U128, U256};
+use web3::types::{BlockId, H256, U128, U256};
 
 use crate::error::DriverError;
 
-use std::env;
-use std::fs;
+use super::base_contract::SmartContract;
 
 type Result<T> = std::result::Result<T, DriverError>;
 
+pub struct SnappContractImpl {
+    base: SmartContract
+}
+
+impl SnappContractImpl {
+    pub fn new(base: SmartContract) -> Self {
+        // Should we assert that the contract is indeed a Snapp contract?
```

Comment:
I think if we wanted to, we should pass no arguments and read the abi file and contract address from env here.

Reply:
Ok. cool!